src/utils/configUtils.ts

```diff
@@ -103,3 +135,149 @@ async function askPreferenceForConfig(configs: IExecutionConfig[], selectedConfi
 export function randomSequence(): string {
     return crypto.randomBytes(3).toString('hex');
 }
+
+type ApplyOperator = (value1: unknown, value2?: unknown) => boolean
+
+interface Token {
+    stringValue: string
+    getValue: () => unknown
+}
+
+interface ResultToken extends Token {
+    getValue: () => boolean
+}
+
+export class WhenClauseEvaluationContext {
+
+    private static readonly OPERATORS: Record<string, ApplyOperator> = {
+        // logical
+        '!': (value: unknown) => !value,
+        '&&': (value1: unknown, value2: unknown) => !!(value1 && value2),
+        '||': (value1: unknown, value2: unknown) => !!(value1 || value2),
+
+        // equality
+        '==': (value1: unknown, value2: unknown) => value1 === value2,
+        '===': (value1: unknown, value2: unknown) => value1 === value2,
+        '!=': (value1: unknown, value2: unknown) => value1 !== value2,
+        '!==': (value1: unknown, value2: unknown) => value1 !== value2,
+
+        // comparison
+        '>': (value1: number, value2: number) => value1 > value2,
+        '>=': (value1: number, value2: number) => value1 >= value2,
+        '<': (value1: number, value2: number) => value1 < value2,
+        '<=': (value1: number, value2: number) => value1 <= value2,
+
+        // match
+        '=~': (value: string, pattern: RegExp) => pattern.test(value),
+    }
+
+    private readonly context: Record<string, unknown> = {};
+
+    public constructor(readonly clause: string) {}
+
+    private tokenize(): Token[] {
+        const tokens: string[] = this.clause.split(/\s+/)
+            .flatMap((token: string) => token.split(/([\(\)]|!(?!=))/))
+            .filter(Boolean);
+
+        return tokens.map((token: string) => ({
+            stringValue: token,
+            getValue: () => this.parse(token),
+        }));
+    }
+
+    private parse(token: string) {
+        const quotedStringMatch: RegExpMatchArray | null = token.match(/['"](.*)['"]/);
+        if (quotedStringMatch)
+            return quotedStringMatch[1];
+
+        const regexMatch: RegExpMatchArray | null = token.match(/\/(?<pattern>.*)\/(?<flags>[ismu]*)/)
+        if (regexMatch?.groups) {
+            const { pattern, flags } = regexMatch.groups;
+            return new RegExp(pattern, flags);
+        }
+
+        const number: number = Number(token);
+        if (!isNaN(number))
+            return number;
+
+        const booleanMatch: RegExpMatchArray | null = token.match(/^true|false$/);
```

Comment:
`/^true|false$/` this will match `^true` or `false$`. I would suggest changing it to `/^(?:true|false)$/`

Reply:
https://github.com/microsoft/vscode-java-test/pull/1638/commits/b231049b1bd33950a32c3d55c7d89c365f729574